external_parser/unit_tests/test_metrics.cc

```diff
@@ -228,4 +228,35 @@ BOOST_AUTO_TEST_CASE(
   should_match_expected_metrics("--ccb_explore_adf", infile_name, outfile_name,
                                 expected_int_metrics, expected_float_metrics,
                                 expected_string_metrics);
+}
+
+BOOST_AUTO_TEST_CASE(
+    check_metrics_ca_mixed_deferred_action_events) {
+  std::string infile_name =
+      "valid_joined_logs/ca_mixed_deferred_action_events_20.log";
+  std::string outfile_name =
+      "ca_mixed_deferred_action_events.json";
+
+  std::map<std::string, int> expected_int_metrics = {
+      {"total_predict_calls", 0},
+      {"total_learn_calls", 10},
+      {"total_log_calls", 233},
```

Comment:
I'd remove the `total_log_calls` because the test might then break if we add/remove log messages and it will be unrelated with what we are testing